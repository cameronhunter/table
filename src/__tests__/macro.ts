import plugin from 'babel-plugin-macros';
import pluginTester from 'babel-plugin-tester';

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import table from '../macro'

      const data = table\`
        cell | presentation   | preloading
        \${2} | \${'slideshow'} | \${true}
        \${3} | \${'static'}    | \${true}
        \${4} | \${'slideshow'} | \${false}
        \${5} | \${'static'}    | \${false}
      \`;
    `
  ]
});
