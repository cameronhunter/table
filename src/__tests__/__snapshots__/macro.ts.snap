// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. unknown plugin: 1. unknown plugin 1`] = `

import table from '../macro'

const data = table\`
  cell | presentation   | preloading
  \${2} | \${'slideshow'} | \${true}
  \${3} | \${'static'}    | \${true}
  \${4} | \${'slideshow'} | \${false}
  \${5} | \${'static'}    | \${false}
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const data = [
  {
    cell: 2,
    presentation: "slideshow",
    preloading: true,
  },
  {
    cell: 3,
    presentation: "static",
    preloading: true,
  },
  {
    cell: 4,
    presentation: "slideshow",
    preloading: false,
  },
  {
    cell: 5,
    presentation: "static",
    preloading: false,
  },
];


`;
